### PR TITLE
fixed various small issues

### PR DESCRIPTION
**Pull request by heejaechang**
fix is for the situation where full analysis option is off.

issue fixed is build + live error de-dup when the option is off. general behavior fix when the option is off.

### CODE REVIEW

**Comment by heejaechang** [comment]
@mavasani can you take a look?

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs [hunk 1]
```diff
@@ -131,7 +137,10 @@ public bool IsEmpty(DocumentId documentId)
 
                 if (!await TryDeserializeDocumentAsync(serializer, document, builder, cancellationToken).ConfigureAwait(false))
                 {
-                    Contract.Requires(false, "How this can happen?");
+                    Contract.Requires(lastResult.Version == VersionStamp.Default);
```
Comment (mavasani) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs]: I presume this fixed https://github.com/dotnet/roslyn/issues/10904? If so, you may want to unskip the new FixAll VSI test that I am adding for v2 engine: https://github.com/dotnet/roslyn-internal/pull/919

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs [hunk 2]
```diff
@@ -399,7 +399,8 @@ protected override bool ShouldIncludeDiagnostic(DiagnosticData diagnostic)
                 // unlike the suppressed (disabled) analyzer, we will include hidden diagnostic only analyzers here.
                 var analyzerDriverOpt = await Owner._compilationManager.CreateAnalyzerDriverAsync(project, stateSets, IncludeSuppressedDiagnostics, cancellationToken).ConfigureAwait(false);
 
-                var result = await Owner._executor.GetProjectAnalysisDataAsync(analyzerDriverOpt, project, stateSets, cancellationToken).ConfigureAwait(false);
+                var ignoreOption = true;
```
Comment (mavasani) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs]: Can you add a comment here and probably rename the flag to ignoreClosedFileDiagnosticsOption?

Comment (mavasani) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs]: Why are we setting this option true only for IDE latest Diagnostic getter? Or is this API only invoked from this specific diagnostic getter?

**Comment by mavasani** [comment]
👍

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs (hunk 1 above)
Comment (heejaechang) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs]: will do.

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs (hunk 2 above)
Comment (heejaechang) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs]: looks like diagnostic getter doesn't care about these options. basically it expects the API to return all results even if full solution analysis is off.

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs (hunk 1 above)
Comment (heejaechang) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs]: @mavasani  how can I enable it?

Comment (mavasani) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs]: I haven't merged the test yet. I can wait for your PR to get merged, and then checkin the test.

**Pull request merged**